### PR TITLE
Add edits and replies to the right panel timeline & prepare the timelineCard to share code with threads

### PR DESCRIPTION
**Pull request by toger5**
Fixes: https://github.com/vector-im/element-web/issues/20012
Fixes: https://github.com/vector-im/element-web/issues/19928

This also lays the groundwork to remove duplicate code between the right panel timeline and the threads view.
the timlineCard(WithComposer) component should have the following properties:
```js
IProps {
    room: Room;
    onClose: () => void;
    resizeNotifier: ResizeNotifier;
    permalinkCreator?: RoomPermalinkCreator;
    e2eStatus?: E2EStatus;
    timelineSet?: EventTimelineSet;
    timelineRenderingType?: TimelineRenderingType;
    showComposer?: boolean;
    composerRelation?: IEventRelation;
}
```
those should be enough to allow the threadsView and the right panel chat to use this component.

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Add edits and replies to the right panel timeline & prepare the timelineCard to share code with threads ([\#7262](https://github.com/matrix-org/matrix-react-sdk/pull/7262)). Fixes vector-im/element-web#20012 and vector-im/element-web#19928. Contributed by @toger5.

Preview: https://61aa264e93ea444beca4ea18--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/right_panel/TimelineCard.tsx [hunk 1]
```diff
@@ -27,77 +28,178 @@ import { Layout } from '../../../settings/enums/Layout';
...
-                    resizeNotifier={this.props.resizeNotifier}
-                    replyToEvent={this.state.replyToEvent}
-                    permalinkCreator={this.props.permalinkCreator}
-                    e2eStatus={this.props.e2eStatus}
-                    compact={true}
-                />
-            </BaseCard>
+            <RoomContext.Provider value={{
+                ...this.context,
+                timelineRenderingType: this.props.timelineRenderingType || TimelineRenderingType.Room,
```
Comment (t3chguy) [inline src/components/views/right_panel/TimelineCard.tsx:157]:
```suggestion
                timelineRenderingType: this.props.timelineRenderingType ?? this.context.timelineRenderingType,
```

IMO would be a saner fallback

File: src/components/views/right_panel/TimelineCard.tsx [hunk 2]
```diff
@@ -27,77 +28,178 @@ import { Layout } from '../../../settings/enums/Layout';
...
+
 }
 interface IState {
     thread?: Thread;
     editState?: EditorStateTransfer;
     replyToEvent?: MatrixEvent;
+    initialEventId?: string;
+    initialEventHighlighted?: boolean;
+
+    // settigns:
```
Comment (t3chguy) [inline src/components/views/right_panel/TimelineCard.tsx:60]:
```suggestion
    // settings:
```

File: src/components/views/right_panel/TimelineCard.tsx [hunk 3]
```diff
@@ -27,77 +28,178 @@ import { Layout } from '../../../settings/enums/Layout';
...
 import { _t } from '../../../languageHandler';
 import { replaceableComponent } from '../../../utils/replaceableComponent';
-
+import { ActionPayload } from '../../../dispatcher/payloads';
+import { Action } from '../../../dispatcher/actions';
+import RoomViewStore from '../../../stores/RoomViewStore';
+import ContentMessages from '../../../ContentMessages';
+import UploadBar from '../../structures/UploadBar';
+import SettingsStore from '../../../settings/SettingsStore';
 interface IProps {
```
Comment (t3chguy) [inline src/components/views/right_panel/TimelineCard.tsx:41]:
```suggestion

interface IProps {
```

File: src/components/views/right_panel/TimelineCard.tsx [hunk 4]
```diff
@@ -27,77 +28,180 @@ import { Layout } from '../../../settings/enums/Layout';
...
 
 @replaceableComponent("structures.TimelineCard")
 export default class TimelineCard extends React.Component<IProps, IState> {
     static contextType = RoomContext;
 
+    private dispatcherRef: string;
+    private timelinePanelRef: React.RefObject<TimelinePanel> = React.createRef();
+    private roomStoreToken: EventSubscription;
+    private settingWatchers: string[];
     constructor(props: IProps) {
```
Comment (t3chguy) [inline src/components/views/right_panel/TimelineCard.tsx:73]:
```suggestion

    constructor(props: IProps) {
```

**Review by t3chguy — APPROVED** [review]
Looks sane to me! :shipit:

File: src/components/views/right_panel/TimelineCard.tsx [hunk 5]
```diff
@@ -27,77 +28,181 @@ import { Layout } from '../../../settings/enums/Layout';
...
     resizeNotifier: ResizeNotifier;
     permalinkCreator?: RoomPermalinkCreator;
     e2eStatus?: E2EStatus;
-    initialEvent?: MatrixEvent;
-    initialEventHighlighted?: boolean;
+    timelineSet?: EventTimelineSet;
+    timelineRenderingType?: TimelineRenderingType;
+    showComposer?: boolean;
+    composerRelation?: IEventRelation;
+
```
Comment (jryans) [inline src/components/views/right_panel/TimelineCard.tsx:52]:
```suggestion
```

**Review by jryans — APPROVED** [review]

**Pull request merged**